client/me/notification-settings/reader-subscriptions/index.jsx

```diff
@@ -238,26 +231,25 @@ const NotificationSubscriptions = createReactClass( {
 						</FormFieldset>
 
 						<FormButton
-							isSubmitting={ this.state.submittingForm }
-							disabled={ this.getDisabledState() }
+							isSubmitting={ this.props.isUpdatingUserSettings }
+							disabled={ this.props.getDisabledState() }
```

Comment:
Should the button also be disabled if there are no changes to save? Right now, the button is enabled when I submit without changes, and the submission stalls because no request is actually performed in that case.

I believe this is also caused by using `this.props.getDisabledState()` instead of `this.getDisabledState()`

Reply:
It's missing a check for `hasUnsavedUserSettings`, I will fix that. Just a note that this is currently the same on prod.

Reply:
Added via https://github.com/Automattic/wp-calypso/pull/48878/commits/ce3238740aafd6f809e47433e4b3e907849064bc